src/library/blas/gens/clTemplates/dot.cl

```diff
@@ -30,7 +30,7 @@ __kernel void %PREFIXdot_kernel( __global %TYPE *_X, __global %TYPE *_Y, __globa
 {
 	__global %TYPE *X = _X + offx;
 	__global %TYPE *Y = _Y + offy;
-    %TYPE dotP = (%TYPE) 0.0;
+    %TYPE dotP = (%TYPE) 0;
```

Comment:
As explained in the commit message, having double precision constants (such as `0.0`) in the source can cause failure on very strict platforms when compiling for devices that do not support double precisions. Beignet on Haswell is one such example. `0.0f` rather than just `0.0` would work too.